src/Illuminate/Auth/Console/stubs/make/tests/Feature/Auth/RegisterTest.stub

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Tests\Feature\Auth;
+
+use App\User;
+use Tests\TestCase;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class RegisterTest extends TestCase
+{
+    use RefreshDatabase;
+
+    protected function fromPage($uri)
+    {
+        return $this->withServerVariables(['HTTP_REFERER' => $uri]);
+    }
+
+    public function testUserCanViewARegistrationForm()
+    {
+        $response = $this->get(route('register'));
+
+        $response->assertSuccessful();
+        $response->assertViewIs('auth.register');
+    }
+
+    public function testUserCannotViewARegistrationFormWhenAuthenticated()
+    {
+        $user = factory(User::class)->make();
+
+        $response = $this->actingAs($user)->get(route('register'));
+
+        $response->assertRedirect(route('home'));
+    }
+
+    public function testUserCanRegister()
+    {
+        $response = $this->post(route('register'), [
+            'name' => 'John Doe',
+            'email' => 'john@example.com',
+            'password' => 'i-love-laravel',
+            'password_confirmation' => 'i-love-laravel',
+        ]);
+
+        $users = User::all();
+        $user = $users->first();
+
+        $response->assertRedirect(route('home'));
+        $this->assertCount(1, $users);
+        $this->assertEquals('John Doe', $user->name);
```

Comment:
I don't see a point. They are strings. And we compare strings. Also, it's not this test's responsibility to check types. The db tests are for that.